tests/multi_block_monitor.rs

```diff
@@ -0,0 +1,185 @@
+    const MAX_DURATION_FOR_SUBMIT_SOLUTION: std::time::Duration =
+        std::time::Duration::from_secs(60 * 20);
+
+    let now = Instant::now();
+
+    let api = loop {
+        if let Ok(api) = ChainClient::from_url("ws://localhost:9966").await {
+            break api;
+        }
+
+        if now.elapsed() > MAX_DURATION_FOR_SUBMIT_SOLUTION {
+            return Err(anyhow::anyhow!("Failed to connect to the API"));
+        }
+
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+    };
+
+    let mut score_submitted = false;
+    let mut pages_submitted = HashSet::new();
+
+    let mut blocks_sub = api.blocks().subscribe_finalized().await?;
+    let pages = api
+        .constants()
+        .at(&runtime::constants().multi_block().pages())
+        .unwrap();
+
+    while let Some(block) = blocks_sub.next().await {
+        if now.elapsed() > MAX_DURATION_FOR_SUBMIT_SOLUTION {
+            break;
+        }
+
+        let block = block?;
+        let events = block.events().await?;
+
+        for ev in events.iter() {
+            let ev = ev?;
+
+            // Score registration.
+            if let Some(item) = ev.as_event::<Registered>()? {
+                if item.1 == alice() {
+                    score_submitted = true;
+                }
+            }
+
+            // Page submission.
+            if let Some(item) = ev.as_event::<Stored>()? {
+                if item.1 == alice() {
+                    pages_submitted.insert(item.0);
+                }
+            }
+
+            if score_submitted && pages_submitted.len() == pages as usize {
+                return Ok(());
+            }
+        }
+    }
+
+    Err(anyhow::anyhow!(
+        "score_submitted: {}, pages_submitted: {:?}; timeout after {}s",
+        score_submitted,
+        pages_submitted,
+        MAX_DURATION_FOR_SUBMIT_SOLUTION.as_secs()
+    ))
+}
+
+async fn run_zombienet() -> KillChildOnDrop {
+    // zombienet --provider native -l text spawn zombienet-staking-runtimes.toml
+    let mut node_cmd = KillChildOnDrop(
+        process::Command::new("zombienet")
+            .stdout(process::Stdio::piped())
+            .stderr(process::Stdio::piped())
+            .args([
+                "--provider",
+                "native",
+                "-l",
+                "text",
+                "spawn",
+                "zombienet-staking-runtimes.toml",
+            ])
+            .env("RUST_LOG", "info,runtime=debug")
+            .spawn()
+            .unwrap(),
+    );
+
+    let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
+    spawn_cli_output_threads(
+        node_cmd.stdout.take().unwrap(),
+        node_cmd.stderr.take().unwrap(),
+        tx,
+    );
+
+    while let Some(line) = rx.recv().await {
+        log::info!("{}", line);
+
+        // We regard this as zombienet started and we can start the miner.
+        if line.to_lowercase().contains("node information") {
```

Comment:
TODO: we could also parse the rpc port from zombienet CLI output but I think it's fine to assume that it start up at port 9966 but if this port is occupied polkadot-sdk will assign a random one.